source/extensions/filters/http/ext_authz/ext_authz.cc

```diff
@@ -44,24 +44,36 @@ void Filter::initiateCall(const Http::RequestHeaderMap& headers) {
   envoy::config::core::v3::Metadata metadata_context;
 
   // If metadata_context_namespaces is specified, pass matching filter metadata to the ext_authz
-  // service.
+  // service. If metadata key is set in both the connection and request metadata then the value
+  // will be the request metadata value.
+  const auto& connection_metadata =
+      decoder_callbacks_->connection()->streamInfo().dynamicMetadata().filter_metadata();
   const auto& request_metadata =
       decoder_callbacks_->streamInfo().dynamicMetadata().filter_metadata();
   for (const auto& context_key : config_->metadataContextNamespaces()) {
-    if (const auto& metadata_it = request_metadata.find(context_key);
+    if (const auto metadata_it = request_metadata.find(context_key);
```

Comment:
why was the reference removed?

Reply:
This was done at the suggestion of wbpcode. https://github.com/envoyproxy/envoy/pull/27612#discussion_r1227462864

I don't have the needed experience with this to pro/con the usage of one or the other.